app/stac_api/models.py

```diff
@@ -197,28 +197,273 @@ def update_geoadmin_variants(self, asset_geoadmin_variant, asset_proj_epsg, asse
-            self.cache_start_datetime = item_properties_start_datetime
+        def insert_new_item():
+            logger.debug('Inserting item %s in collection %s', item_id, self.collection_name)
+            if self.cache_start_datetime is None:
+                # first item in collection, as cache_start_datetime is None:
+                self.cache_start_datetime = new_start_datetime
+            elif self.cache_start_datetime > new_start_datetime:
+                # new item starts earlier that current collection range starts
+                self.cache_start_datetime = new_start_datetime
+
+            if self.cache_end_datetime is None:
+                #first item in collection, as cache_start_datetime is None
+                self.cache_end_datetime = new_end_datetime
+            elif self.cache_end_datetime < new_end_datetime:
+                # new item starts after current collection's range ends
+                self.cache_end_datetime = new_end_datetime
+
             self.save()
-        elif self.cache_end_datetime is None:
-            self.cache_end_datetime = item_properties_end_datetime
+
+        def update_item():
+            logger.debug('Updating item %s in collection %s', item_id, self.collection_name)
+            if old_start_datetime != new_start_datetime:
+                # start_datetime was updated
+                if old_start_datetime == self.cache_start_datetime:
+                    # item's old start_datetime was defining left bound of the temporal
+                    # extent interval of collection before update
+                    if new_start_datetime < old_start_datetime:
+                        # item's start_datetime was shifted to the left (earlier)
+                        self.cache_start_datetime = new_start_datetime
+                    else:
+                        # item's start_datetime was shifted to the right (later)
+                        # but was defining the left bound of the temporal extent
+                        # of the collection before
+                        # --> hence the new start_datetime of the collection
+                        # needs to be determinded:
+                        # get all items but the one to that is beeing updated:
+                        logger.warning(
+                            'Looping (twice) over all items of collection %s,'
+                            'to update temporal extent, this may take a while',
+                            self.collection_name
+                        )
+                        qs = Item.objects.filter(collection_id=self.pk).exclude(id=item_id)
+                        # get earliest start_datetime or none, in case none exists
+                        # this should be the case, if the item that is currently
+                        # beeing updated is the only item in the collection or
+                        # no item has a range defined (no start_ and end_datetimes)
```

Comment:
using `bool()` here, due to this:
"if a some_queryset has not yet been evaluated, but you know that it will be at some point, then using some_queryset.exists() will do more overall work (one query for the existence check plus an extra one to later retrieve the results) than using bool(some_queryset), which retrieves the results and then checks if any were returned."
(see: https://docs.djangoproject.com/en/3.0/ref/models/querysets/)